### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by PhantomKojima** [ParisNeo/lollms-webui#444] [Q&A]
1. Using the HuggingFace binding, I can’t find gguf models in the search. What binding do I need to use gguf models on nvidia gpu?
2. The search for models does not work until I click the "Add reference" button in Add models for binding. Is this how it should be?
3. Are all the files in the installer_files folder necessary to use the program? It turns out to be not the most compact program. Can the program be used as a portable version?
4. If I try to add a .TXT file instead of .txt, I will get an error stating that the format is not supported. It probably works as it should, but it would be nice if .TXT files were supported too. I haven’t checked, but probably the same problem exists with PDF files and others.
5. Sometimes the program stops working after warmup for generating up to ____ tokens, what should I do about it?

**Comment by ParisNeo** [comment]
1- GGUF models are llamacpp models (like ggml), these are not models that run on a typical hugging face transformers model. They are supported through other bindings like cTransformers and gpt4all bindings.
2- The models are loaded using lazy loading because otherwize it will be very slow. IIf you search a model, first type its name then press enter to update the list
3- lollms uses lots of libraries under the hood. It is a giant tool after all that tries to be compatible with lots of technologies and literally builds an entire python environment. For example, when you install it it will install cuda libraries to comile some bindings and libraries. It also installs many other libraries that are mandatory to its inner workings. It reinstalls conda as well as python. This is a design choice to make the application 100% self sufficient and doesn't interfere with your own python libraries. You can literally delete everything by deleting the folder. Keep in mind that the AI can write and execute code in lollms so I am obliged to preinstall several libraries. The AI will install other libraries if it needs to. This is not automatic ofcourse, but some personas do have autonomy and you need to be careful with them and read the disclaimer that appears when you mount a persona. 
4- Good point. I just fixed that. you should just rerun the program and it should work fine if you have autoupdate active (it is the default setting).
5- That's not an expected outcome. Try press cancel generation then retry again.

**Reply by PhantomKojima** [reply]
3. Yes, I can delete the folder but then I will just get the message "Miniconda not found. Please reinstall lollms using win_install.bat." so the program will not load miniconda by itself.
4. 
5. When I say that the program stops working, I mean that in cmd I get the line Press any key to continue... and the UI stops working. So I changed the binding to CTransformers and it seemed to work fine. Apparently gpt4all binding is very unstable on my computer.
6. And I can’t find documentation on the program, what libraries it uses, etc. For example, I want to find out what files the model can read, I could not find this information and found out the list of supported files in cmd only when I loaded an unsupported one. Where can I find a list of supported image files? Update: Ok, I found the documentation here https://parisneo.github.io/lollms-webui/ but it is not complete.
7. I asked lollms if he could create a txt file and write a word in it. It said it would do it, but in fact it didn't. Is the model simply unaware of its capabilities?
8. Now I'm using CTransformers binding and a gguf model with the same number of GPU layers as in LM Studio and lollms is much slower. Can I do something about this?
9. Where does lollms store its chat cache?
10. How does long term memory work?
11. I understand that "warmup for generating up to 124 tokens" means that the next message cannot be longer than 124 tokens, can I somehow change this?
12. It would be great to see what file I attached to the message so I can refer to it in the future.
13. If I attach an image, a message with it is immediately sent and lollms gives me its description, while the image remains in the list of attached files in the current message, is this how it should be? And what if I don't need a description of this image, for example I just need to use it for something like in the video from your channel "From Sketch to App: Unleashing the Power of LoLLMs' Vision Capability"? Again, it would be nice to be able to reference this particular image in the future.
14. How do personalities work? For example, I will choose the Music Lyrics Expert personality, the selected model will not know the lyrics of the songs better, then what is the point?
15. Sometimes I get a message in cmd "D:\Prgrms\LOLLMSV7.0\installer_files\lollms_env\lib\site-packages\ctransformers\llm.py:459: UserWarning: `LLM.reset()` method is deprecated since 0.2.27 "Please use high-level API." I don't know if this is important.
16. In the personality description, I saw a mention of lollms personality maker but could not find information on this on the Internet. So what is it? Can i create a new personality myself?
sorry for so many questions
